File: software/examples/rust/example_touch.rs
use std::{error::Error, io, thread};
use tinkerforge::{ip_connection::IpConnection, lcd_128x64_bricklet::*};

const HOST: &str = "localhost";
const PORT: u16 = 4223;
const UID: &str = "XYZ"; // Change XYZ to the UID of your LCD 128x64 Bricklet.

fn main() -> Result<(), Box<dyn Error>> {
    let ipcon = IpConnection::new(); // Create IP connection.
    let lcd = Lcd128x64Bricklet::new(UID, &ipcon); // Create device object.

    ipcon.connect((HOST, PORT)).recv()??; // Connect to brickd.
                                          // Don't use device before ipcon is connected.

    let touch_position_receiver = lcd.get_touch_position_callback_receiver();

    // Spawn thread to handle received callback messages.
    // This thread ends when the `lcd` object
    // is dropped, so there is no need for manual cleanup.
    thread::spawn(move || {
        for touch_position in touch_position_receiver {
            println!("Pressure: {}", touch_position.pressure);
            println!("X: {}", touch_position.x);
            println!("Y: {}", touch_position.y);
            println!("Age: {}", touch_position.age);
            println!();
        }
    });

    let touch_gesture_receiver = lcd.get_touch_gesture_callback_receiver();

    // Spawn thread to handle received callback messages.
    // This thread ends when the `lcd` object
    // is dropped, so there is no need for manual cleanup.
    thread::spawn(move || {
        for touch_gesture in touch_gesture_receiver {
            if touch_gesture.gesture == LCD_128X64_BRICKLET_GESTURE_LEFT_TO_RIGHT {
                println!("Gesture: Left To Right");
            } else if touch_gesture.gesture == LCD_128X64_BRICKLET_GESTURE_RIGHT_TO_LEFT {
                println!("Gesture: Right To Left");
            } else if touch_gesture.gesture == LCD_128X64_BRICKLET_GESTURE_TOP_TO_BOTTOM {
                println!("Gesture: Top To Bottom");
            } else if touch_gesture.gesture == LCD_128X64_BRICKLET_GESTURE_BOTTOM_TO_TOP {
                println!("Gesture: Bottom To Top");
            }

            println!("Duration: {}", touch_gesture.duration);
            println!("Pressure Max: {}", touch_gesture.pressure_max);
            println!("X Start: {}", touch_gesture.x_start);
            println!("X End: {}", touch_gesture.x_end);
            println!("Y Start: {}", touch_gesture.y_start);
            println!("Y End: {}", touch_gesture.y_end);
            println!("Age: {}", touch_gesture.age);
            println!();
        }
    });

    // Set period for touch position callback to 0.1s (100ms).
    lcd.set_touch_position_callback_configuration(100, true);

    // Set period for touch gesture callback to 0.1s (100ms).
    lcd.set_touch_gesture_callback_configuration(100, true);

    println!("Press enter to exit.");
    let mut _input = String::new();
    io::stdin().read_line(&mut _input)?;
    ipcon.disconnect();
    Ok(())
}

File: software/examples/go/example_touch.go
package main

import (
	"fmt"
	"github.com/tinkerforge/go-api-bindings/ipconnection"
	"github.com/tinkerforge/go-api-bindings/lcd_128x64_bricklet"
)

const ADDR string = "localhost:4223"
const UID string = "XYZ" // Change XYZ to the UID of your LCD 128x64 Bricklet.

func main() {
	ipcon := ipconnection.New()
	defer ipcon.Close()
	lcd, _ := lcd_128x64_bricklet.New(UID, &ipcon) // Create device object.

	ipcon.Connect(ADDR) // Connect to brickd.
	defer ipcon.Disconnect()
	// Don't use device before ipcon is connected.

	lcd.RegisterTouchPositionCallback(func(pressure uint16, x uint16, y uint16, age uint32) {
		fmt.Printf("Pressure: %d\n", pressure)
		fmt.Printf("X: %d\n", x)
		fmt.Printf("Y: %d\n", y)
		fmt.Printf("Age: %d\n", age)
		fmt.Println()
	})

	lcd.RegisterTouchGestureCallback(func(gesture lcd_128x64_bricklet.Gesture, duration uint32, pressureMax uint16, xStart uint16, xEnd uint16, yStart uint16, yEnd uint16, age uint32) {

		if gesture == lcd_128x64_bricklet.GestureLeftToRight {
			fmt.Println("Gesture: Left To Right")
		} else if gesture == lcd_128x64_bricklet.GestureRightToLeft {
			fmt.Println("Gesture: Right To Left")
		} else if gesture == lcd_128x64_bricklet.GestureTopToBottom {
			fmt.Println("Gesture: Top To Bottom")
		} else if gesture == lcd_128x64_bricklet.GestureBottomToTop {
			fmt.Println("Gesture: Bottom To Top")
		}

		fmt.Printf("Duration: %d\n", duration)
		fmt.Printf("Pressure Max: %d\n", pressureMax)
		fmt.Printf("X Start: %d\n", xStart)
		fmt.Printf("X End: %d\n", xEnd)
		fmt.Printf("Y Start: %d\n", yStart)
		fmt.Printf("Y End: %d\n", yEnd)
		fmt.Printf("Age: %d\n", age)
		fmt.Println()
	})

	// Set period for touch position callback to 0.1s (100ms).
	lcd.SetTouchPositionCallbackConfiguration(100, true)

	// Set period for touch gesture callback to 0.1s (100ms).
	lcd.SetTouchGestureCallbackConfiguration(100, true)

	fmt.Print("Press enter to exit.")
	fmt.Scanln()

}
